src/scripts/components/DropdownMultiSelectAdvanced.jsx

```diff
@@ -0,0 +1,246 @@
+    }
+
+    return null;
+  };
+
+  renderUser = (id, idx) => {
+    const user = this.props.selectedUsers[id];
+    return this.renderUserResource(user, id, idx);
+  };
+
+  renderViewSelectedUsers = () => (
+    <Dropdown wide onClick={this.clearSearch} autoFocusInput={false} label={this.props.dropdownLabel} type="outline-primary" disableScroll>
+      <div className="dropdown__menu__container">
+        <div className="search__group">
+          <UtilityInlineGrid className="u-flex u-flex-justify-between u-m-t-small u-text-small">
+            <Button
+              size="small"
+              type="link"
+              onClick={this.handleClearAll}
+              title="Clear All"
+            >Clear All
+            </Button>
+            <div>
+              <Button
+                size="small"
+                type="link"
+                onClick={this.handleToggle}
+                title="Back"
+              >Back
+              </Button>
+            </div>
+          </UtilityInlineGrid>
+        </div>
+        {this.props.selectedUserIds.length > 0 ? (
+          <Scrollbars className="resource-group__scroll" autoHeight autoHeightMax={UtilitySystem.config.resourceSizes.large}>
+            <ResourceGroup interfaceMode="checkbox">
+              {this.props.selectedUserIds.sort(this.sortMemberSearchIds).map(this.renderUser)}
+            </ResourceGroup>
+          </Scrollbars>
+        ) :
+          'No Users Added'
+        }
+      </div>
+    </Dropdown>
+  );
+
+  render() {
+    const { userSearchLoading, openPanel, dropdownLabel, selectedUserIds, filterName } = this.props;
+    const usersIds = [...this.props.usersIds];
+    const title = `View Selected (${this.props.selectedUserIds.length})`;
+    const searchTitle = `Search ${filterName}`;
+    let returnValue = '';
+    if (openPanel) {
+      returnValue = (
+        this.state.viewAllUsers ?
+          (
+            <Dropdown wide autoFocusInput={false} label={dropdownLabel} onClick={this.clearSearch} type="outline-primary" disableScroll>
+              <div className="dropdown__menu__container">
+                <div className="search__group">
+                  {selectedUserIds.length > 0 ? (
+                    <UtilityInlineGrid className="u-flex u-flex-justify-between u-m-t-small u-text-small">
+                      <Button
+                        size="small"
+                        type="link"
+                        onClick={this.handleClearAll}
+                        title="Clear All"
+                      >Clear All
+                      </Button>
+                      <Button
+                        size="small"
+                        type="link"
+                        onClick={this.handleToggle}
+                        disabled={this.props.selectedUserIds.length === 0}
+                        title={title}
+                      >{title}
+                      </Button>
+                    </UtilityInlineGrid>)
```

Comment:
I am seeing this code is repeated on more that one place. Wrap this code in a separate function if possible and use that function.